### DISCUSSION TITLE
How can I track different streamlines in DIPY?

### DISCUSSION THREAD

**Discussion by Redouan20102** [dipy/dipy#2222] [Tractography / Tractometry] [answered]
I'm working on a project where i have to use DIPY to track different streamlines of the brain and show them separately, I found the code to track the Corpus Callosum on DIPY's official website. Only I couldn't understand how to make it track other parts like the Cingulum for instance. 
anyone has any idea on how to do that? 
thank you all.

**Comment by arokem** [comment]
You can use the DIPY CLI to call the [Recobundles](https://pubmed.ncbi.nlm.nih.gov/28712994/) algorithm. There's an example of the sequence of CLI calls here: https://github.com/dipy/dipy_qa/blob/master/code/simple_flow_series.sh

Alternatively, if you are feeling adventurous, you can use the DIPY-dependent project [pyAFQ](http://yeatmanlab.github.io/pyAFQ/), which tries to emulate the [AFQ](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0049790) algorithm, but note that this project is still in alpha status.

**Comment by skoudoro** [comment]
To complete @arokem information:

- More details on basic the CLI on this page: https://dipy.org/documentation/1.2.0./interfaces/basic_flow/ 
- Current available tutorial on CLI : https://dipy.org/documentation/1.2.0./interfaces
- the list of CLI : https://dipy.org/documentation/1.2.0./reference_cmd/

**Comment by Redouan20102** [comment, marked as answer]
thank you @skoudoro and @arokem for your answers, but my question ws much simpler than that. I believe I have found the answer inside the dataset files.
I'm working with the Stanford-HARDI dataset and I found a file names "label_info.txt" inside of it today with labels for each bundle of the white matter, here it is in case anyone didn't find it: 

> new label, freesurfef label, freesurfer name
> 1, 2, "Left-Cerebral-White-Matter"
> 1, 41, "Right-Cerebral-White-Matter"
> 1, 77, "WM-hypointensities"
> 1, 85, "Optic-Chiasm"
> 1, 1004, "ctx-lh-corpuscallosum"
> 1, 2004, "ctx-rh-corpuscallosum"
> 2, 251, "CC_Posterior"
> 2, 252, "CC_Mid_Posterior"
> 2, 253, "CC_Central"
> 2, 254, "CC_Mid_Anterior"
> 2, 255, "CC_Anterior"
> 3, 1032, "ctx-lh-frontalpole"
> 4, 1014, "ctx-lh-medialorbitofrontal"
> 5, 1012, "ctx-lh-lateralorbitofrontal"
> 6, 1019, "ctx-lh-parsorbitalis"
> 7, 1020, "ctx-lh-parstriangularis"
> 8, 1018, "ctx-lh-parsopercularis"
> 9, 1027, "ctx-lh-rostralmiddlefrontal"
> 10, 1003, "ctx-lh-caudalmiddlefrontal"
> 11, 1028, "ctx-lh-superiorfrontal"
> 12, 1024, "ctx-lh-precentral"
> 13, 1017, "ctx-lh-paracentral"
> 14, 1035, "ctx-lh-insula"
> 15, 1026, "ctx-lh-rostralanteriorcingulate"
> 16, 1002, "ctx-lh-caudalanteriorcingulate"
> 17, 1023, "ctx-lh-posteriorcingulate"
> 18, 1010, "ctx-lh-isthmuscingulate"
> 19, 1006, "ctx-lh-entorhinal"
> 20, 1007, "ctx-lh-fusiform"
> 21, 1016, "ctx-lh-parahippocampal"
> 22, 1009, "ctx-lh-inferiortemporal"
> 23, 1033, "ctx-lh-temporalpole"
> 24, 1015, "ctx-lh-middletemporal"
> 25, 1030, "ctx-lh-superiortemporal"
> 26, 1034, "ctx-lh-transversetemporal"
> 27, 1001, "ctx-lh-bankssts"
> 28, 1022, "ctx-lh-postcentral"
> 29, 1031, "ctx-lh-supramarginal"
> 30, 1008, "ctx-lh-inferiorparietal"
> 31, 1029, "ctx-lh-superiorparietal"
> 32, 1025, "ctx-lh-precuneus"
> 33, 1005, "ctx-lh-cuneus"
> 34, 1011, "ctx-lh-lateraloccipital"
> 35, 1021, "ctx-lh-pericalcarine"
> 36, 1013, "ctx-lh-lingual"
> 37, 11, "Left-Caudate"
> **38, 12, "Left-Putamen"
> **39, 13, "Left-Pallidum"
> 40, 9, "Left-Thalamus"
> 41, 10, "Left-Thalamus-Proper"
> 42, 17, "Left-Hippocampus"
> 43, 18, "Left-Amygdala"
> 44, 26, "Left-Accumbens-area"
> **45, 28, "Left-VentralDC"
> 46, 2032, "ctx-rh-frontalpole"
> 47, 2014, "ctx-rh-medialorbitofrontal"
> 48, 2012, "ctx-rh-lateralorbitofrontal"
> 49, 2019, "ctx-rh-parsorbitalis"
> 50, 2020, "ctx-rh-parstriangularis"
> 51, 2018, "ctx-rh-parsopercularis"
> 52, 2027, "ctx-rh-rostralmiddlefrontal"
> 53, 2003, "ctx-rh-caudalmiddlefrontal"
> 54, 2028, "ctx-rh-superiorfrontal"
> 55, 2024, "ctx-rh-precentral"
> 56, 2017, "ctx-rh-paracentral"
> 57, 2035, "ctx-rh-insula"
> 58, 2026, "ctx-rh-rostralanteriorcingulate"
> 59, 2002, "ctx-rh-caudalanteriorcingulate"
> 60, 2023, "ctx-rh-posteriorcingulate"
> 61, 2010, "ctx-rh-isthmuscingulate"
> 62, 2006, "ctx-rh-entorhinal"
> 63, 2007, "ctx-rh-fusiform"
> 64, 2016, "ctx-rh-parahippocampal"
> 65, 2009, "ctx-rh-inferiortemporal"
> 66, 2033, "ctx-rh-temporalpole"
> 67, 2015, "ctx-rh-middletemporal"
> 68, 2030, "ctx-rh-superiortemporal"
> 69, 2034, "ctx-rh-transversetemporal"
> 70, 2001, "ctx-rh-bankssts"
> 71, 2022, "ctx-rh-postcentral"
> 72, 2031, "ctx-rh-supramarginal"
> 73, 2008, "ctx-rh-inferiorparietal"
> 74, 2029, "ctx-rh-superiorparietal"
> 75, 2025, "ctx-rh-precuneus"
> 76, 2005, "ctx-rh-cuneus"
> 77, 2011, "ctx-rh-lateraloccipital"
> 78, 2021, "ctx-rh-pericalcarine"
> 79, 2013, "ctx-rh-lingual"
> 80, 50, "Right-Caudate"
> 81, 51, "Right-Putamen"
> 82, 52, "Right-Pallidum"
> 83, 48, "Right-Thalamus"
> 84, 49, "Right-Thalamus-Proper"
> 85, 53, "Right-Hippocampus"
> 86, 54, "Right-Amygdala"
> 87, 58, "Right-Accumbens-area"
> 88, 60, "Right-VentralDC"


thanks again for your help, if you ave nothing to add I'll close the issue.

**Comment by skoudoro** [comment]
oh ok, that's great! nothing to add 😄. 

I am closing this issue. Thank you for your feedback @Redouan20102!
